tests/testDegenerateTemplate.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+#
+# Copyright 2008-2016  AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import print_function
+import unittest
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.detection as afwDet
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+import lsst.pex.logging as pexLogging
+from lsst.meas.deblender.baseline import deblend
+import lsst.meas.algorithms as measAlg
+
+root = pexLogging.Log.getDefaultLog()
+root.setThreshold(pexLogging.Log.DEBUG)
+
+# Quiet some of the more chatty loggers
+pexLogging.Log(root, 'lsst.meas.deblender.symmetrizeFootprint',
+               pexLogging.Log.INFO)
+pexLogging.Log(root, 'lsst.meas.deblender.symmetricFootprint',
+               pexLogging.Log.INFO)
+pexLogging.Log(root, 'lsst.meas.deblender.getSignificantEdgePixels',
+               pexLogging.Log.INFO)
+pexLogging.Log(root, 'afw.Mask',
+               pexLogging.Log.INFO)
+
+
+def imExt(img):
+    bbox = img.getBBox()
+    return [bbox.getMinX(), bbox.getMaxX(),
+            bbox.getMinY(), bbox.getMaxY()]
+
+
+def doubleGaussianPsf(W, H, fwhm1, fwhm2, a2):
+    return measAlg.DoubleGaussianPsf(W, H, fwhm1, fwhm2, a2)
+
+
+def gaussianPsf(W, H, fwhm):
+    return measAlg.DoubleGaussianPsf(W, H, fwhm)
+
+
+class DegenerateTemplateTestCase(lsst.utils.tests.TestCase):
+
+    def test1(self):
+        '''
+        A simple example: three overlapping blobs (detected as 1
+        footprint with three peaks).  Additional peaks are added near
+        the blob peaks that should be identified as degenerate.
+        '''
+        H, W = 100, 100
+
+        fpbb = afwGeom.Box2I(afwGeom.Point2I(0, 0),
+                             afwGeom.Point2I(W-1, H-1))
```

Comment:
Spacing around -'s. 
Don't really need the carriage return (full line still < 110).